pkg/apis/options/providers.go

```diff
@@ -0,0 +1,104 @@
+	GitLabConfig    GitLabOptions    `json:"gitlabConfig,omitempty"`
+	GoogleConfig    GoogleOptions    `json:"googleConfig,omitempty"`
+	OIDCConfig      OIDCOptions      `json:"oidcConfig,omitempty"`
+	LoginGovConfig  LoginGovOptions  `json:"loginGovConfig,omitempty"`
+
+	ProviderID      string   `json:"providerID,omitempty"`
+	ProviderType    string   `json:"provider,omitempty"`
+	ProviderName    string   `json:"providerDisplayName,omitempty"`
+	ProviderCAFiles []string `json:"providerCAFiles,omitempty"`
+
+	LoginURL          string   `json:"loginURL,omitempty"`
+	RedeemURL         string   `json:"redeemURL,omitempty"`
+	ProfileURL        string   `json:"profileURL,omitempty"`
+	ProtectedResource string   `json:"resource,omitempty"`
+	ValidateURL       string   `json:"validateURL,omitempty"`
+	Scope             string   `json:"scope,omitempty"`
+	Prompt            string   `json:"prompt,omitempty"`
+	ApprovalPrompt    string   `json:"approvalPrompt,omitempty"`
+	AllowedGroups     []string `json:"allowedGroups,omitempty"`
+
+	AcrValues string `json:"acrValues,omitempty"`
+}
+
+type KeycloakOptions struct {
+	KeycloakGroup string `json:"keycloakGroup,omitempty"`
+}
+
+type AzureOptions struct {
+	AzureTenant string `json:"azureTenant,omitempty"`
+}
+
+type BitbucketOptions struct {
+	BitbucketTeam       string `json:"bitbucketTeam,omitempty"`
+	BitbucketRepository string `json:"bitbucketRepository,omitempty"`
+}
+
+type GitHubOptions struct {
+	GitHubOrg   string   `json:"githubOrg,omitempty"`
+	GitHubTeam  string   `json:"githubTeam,omitempty"`
+	GitHubRepo  string   `json:"githubRepo,omitempty"`
+	GitHubToken string   `json:"githubToken,omitempty"`
+	GitHubUsers []string `json:"githubUsers,omitempty"`
+}
+
+type GitLabOptions struct {
+	GitLabGroup []string `json:"gitlabGroups,omitempty"`
+}
+
+type GoogleOptions struct {
+	GoogleGroups             []string `json:"googleGroup,omitempty"`
+	GoogleAdminEmail         string   `json:"googleAdminEmail,omitempty"`
+	GoogleServiceAccountJSON string   `json:"googleServiceAccountJson,omitempty"`
+}
+
+type OIDCOptions struct {
+	OIDCIssuerURL                      string `json:"oidcIssuerURL,omitempty"`
+	InsecureOIDCAllowUnverifiedEmail   bool   `json:"insecureOidcAllowUnverifiedEmail,omitempty"`
+	InsecureOIDCSkipIssuerVerification bool   `json:"insecureOidcSkipIssuerVerification,omitempty"`
+	SkipOIDCDiscovery                  bool   `json:"skipOidcDiscovery,omitempty"`
+	OIDCJwksURL                        string `json:"oidcJwksURL,omitempty"`
+	OIDCGroupsClaim                    string `json:"oidcGroupsClaim,omitempty"`
+	UserIDClaim                        string `json:"userIDClaim,omitempty"`
+}
+
+type LoginGovOptions struct {
+	JWTKey     string `json:"jwtKey,omitempty"`
+	JWTKeyFile string `json:"jwtKeyFile,omitempty"`
+	PubJWKURL  string `json:"pubjwkURL,omitempty"`
+}
+
+func providerDefaults() Providers {
+	providers := Providers{
+		Provider{
```

Comment:
Nit: Pretty sure you don't need to specify the type here and can just
```suggestion
		{
```